apps/arweave/src/ar_mining_io.erl

```diff
@@ -242,51 +248,105 @@ get_packed_intervals(Start, End, MiningAddress, "default", Intervals) ->
 get_packed_intervals(_Start, _End, _ReplicaID, _StoreID, _Intervals) ->
 	no_interval_check_implemented_for_non_default_store.
 
-filter_by_packing([], _Intervals, _StoreID) ->
-	[];
-filter_by_packing([{EndOffset, Chunk} | ChunkOffsets], Intervals, "default" = StoreID) ->
-	case ar_intervals:is_inside(Intervals, EndOffset) of
-		false ->
-			filter_by_packing(ChunkOffsets, Intervals, StoreID);
+maybe_clear_cached_chunks(Cache, LastClearTime) ->
+	Now = os:system_time(millisecond),
+	case (Now - LastClearTime) > (?CACHE_TTL_MS div 2) of
```

Comment:
Mostly to limit the impact of sampling frequency. We want to check the cache for obsolete data more frequently than we expire that data. e.g. if I only check the cache every 2seconds, it's possible a cache entry could persist up to 4 seconds in the cache (e.g. if a cache entry was added right after we check the cache it won't have expired by the time we check the cache again 2 seconds later, and so will only be removed when check yet again another 2 seconds after that - for almost 4 seconds total)

Byt checking ever 1 second and then only evicting items that are 2 seconds old, we keep the true TTL closer to 2 seconds (max of 3 seconds)

Checking the cache is actually a pretty quick operations so checking it every second didn't seem to impact much

Reply:
maybe move `maybe_clear_cached_chunks` after `send_chunks`?

Reply:
done!

Reply:
After this move. Do we still need `div 2)`?

Reply:
yes I believe they are unrelated. We always call `maybe_clear_cached_chunks` but we don't alwayscheck the cache. So moving the call to maybe_clear_cached_chunks doesn't impact the sampling frequency - we still want to check the cache more frequently than the TTL to limit the aliasing which can cause ranges to remain in the cache longer than the TTL

if anything I might want to make it div 4 or even higher. But so far in my tests having ranges remain in the cache from 2-3 seconds seems to be fine. If it ends up being too long, I might decrease teh TTL *and* increase the sampling frequency

Reply:
I think div 2 will not make difference
`receive` is not time sampled, it's request sampled. So time intervals can have really big gaps

Reply:
What's the problem with leaving it in? I've tested the version of teh code with the div 2 and it works as expected. Earlier versions of the code without the div 2 had larger memory footprints. Receive may not be time sampled, but we don't know how frequently a request comes it - but it may often be roughly once per second due to VDF -  and since we want to keep data in cache for as close to the TTL as possible checking teh cache more frequently is better than checking it less frequently 

I don't konw what we gain by removing div 2, but my testing so far indicates that it does serve a purpose